sdks/java/container/boot.go

```diff
@@ -153,24 +153,19 @@ func main() {
 			cp = append(cp, filepath.Join(jarsDir, jar))
 		}
 	}
-	cp = append(cp, filepath.Join(jarsDir, javaHarnessJar))
-
 	var hasWorkerExperiment = strings.Contains(options, "use_staged_dataflow_worker_jar")
+
+	if !hasWorkerExperiment {
```

Comment:
nit: it might read a bit more easily if you reverse the branches (positive, then negative)